### ISSUE TITLE
Loss function in VAE

### ISSUE THREAD

**Issue by bcpenggh** [2017-fall-DL-training-program/VAE-GAN-and-VAE-GAN#25]
Dear TA,

In the example code of VAE, the loss function is the summation of cross entropy and KL divergence.
![loss_fun_vae](https://user-images.githubusercontent.com/32902010/33815246-d985d340-de6a-11e7-8adf-e371562f98b2.jpg)

On the other hand in VAE-GAN, the loss function is the summation of MSE and KL divergence.
![loss_fun_vae-gan](https://user-images.githubusercontent.com/32902010/33815316-58de7458-de6b-11e7-8a30-020433dfc08a.jpg)

Both of them seems work well. Are they equivalent? And which one matches the equation in the handout?
![loss_fun_eqn](https://user-images.githubusercontent.com/32902010/33815380-e5e4ac32-de6b-11e7-9fe2-d27c6ba3eefc.jpg)

**Comment by hui-po-wang** [comment]
Hi @bcpenggh,

The former term is a likelihood term. You can choose any distribution as your "belief". For example, I can "believe" that my likelihood should follow a Gaussian distribution with N(o1(x), sigma^2 * I) where  o1 is unknown parameters and approximated by a neural network. Then, based on the idea of "Maximum Likelihood (MLE)", we should try to maximize the probability that the original training data appears. Note that our likelihood now is believed to be a "Gaussian". According to the formulation, 

![default](https://user-images.githubusercontent.com/10228478/33816315-03f9fba8-de73-11e7-8b34-396ba83e1d57.png)

, to maximize the probability, our prediction should be as close as possible to the training data. That is, we should minimize the MSE between the prediction and the training data.

A similar story can be applied to BCE.

By the way, a likelihood function, p(x|z), means given an observation z how much likely an even, x, will happen.

Thanks
